### ISSUE TITLE
H9: DQD-World - Differentiable Quality Diversity for World Model Optimization

### ISSUE THREAD

**Issue by pranavp311** [justrach/agentlab#16]
## Research Hypothesis 9 (Inspired by Clune's DQD)

**Core Idea:** Apply Differentiable Quality Diversity (DQD) - specifically CMA-MEGA/OMG-MEGA - to optimize world model parameters while maintaining diversity across latent space structure.

**Motivation:**
- Fontaine & Nikolaidis (NeurIPS 2021) introduced DQD which uses gradients to dramatically improve QD efficiency
- World models are differentiable - perfect for DQD
- Can illuminate the space of "how world models compress observations" while maintaining prediction quality

**Research Questions:**
1. Can we define differentiable behavior descriptors for world model training?
   - E.g., latent space disentanglement, sparsity, clustering structure
2. Does DQD find world models faster than random/grid search for hyperparameters?
3. Can gradient-based QD discover Pareto-optimal trade-offs between reconstruction and prediction?

**DQD-World Algorithm:**
```python
# Objective: World model prediction loss (differentiable)
F(θ) = E[||s_{t+1} - f_θ(s_t, a_t)||^2]

# Behavior descriptors (differentiable):
m1(θ) = latent_space_disentanglement(z)  # via gradient
m2(θ) = reconstruction_quality(x, x_hat)  # via gradient

# CMA-MEGA update:
# Use gradient arborescence to branch in objective-measure space
θ_new = θ + α·∇F + β·∇m1 + γ·∇m2
where α, β, γ ~ N(μ, Σ) adapted by CMA-ES
```

**Key Innovations:**
- First application of DQD to world model training
- Gradient-based illumination of world model design space
- Efficient Pareto front approximation for model objectives

**Advantages over standard hyperparameter search:**
- 10-100x more sample efficient (from DQD paper results)
- Returns diverse set of models, not just one "best"
- Reveals trade-offs between competing objectives

**Key Papers:**
- Fontaine & Nikolaidis (NeurIPS 2021) - Differentiable Quality Diversity
- Fontaine et al. (2020) - CMA-ME
- Ha & Schmidhuber (2018) - World Models

**Feasibility: HIGH** - Well-defined scope, builds on solid theory
**Compute Requirements: MEDIUM** - Gradients make it efficient
